### PR TITLE
Events

### PR DESCRIPTION
**Pull request by Ashah15**

### CODE REVIEW

File: private-events/app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  # Returns true if the given token matches the digest.
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
```
Comment (maelfosso) [inline private-events/app/models/user.rb]: Why that line please?

File: private-events/app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    def show
+        @user = current_user
+    	@event = Event.find(params[:id])
+        @attendees = @event.attendees
+        @isAttendee = @event.attendees.exists?(:name => current_user.username)
+    end
+
+    def index
+    	@user = current_user
```
Comment (maelfosso) [inline private-events/app/controllers/events_controller.rb]: Why do you need that?

File: private-events/app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		if @user.save
+			log_in @user
+    	redirect_to @user
+		else
+			 render 'new'
+		end
+	end
+
+	def show
+		@user = current_user
```
Comment (maelfosso) [inline private-events/app/controllers/users_controller.rb]: Why that line?

File: private-events/app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		@upcoming_events = Event.upcoming.joins(:attendees).where(attendees: { name: current_user.username })
+		@past_events = Event.past.joins(:attendees).where(attendees: { name: current_user.username })
```
Comment (maelfosso) [inline private-events/app/controllers/users_controller.rb]: Please, I suggest you use `@current_user.attendees.upcoming`.

File: private-events/app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    	params.require(:user).permit(:username, :email)
+    end
+
+    def logged_in_user
+      unless logged_in?
+        store_location
+        flash[:danger] = "Please log in."
+        redirect_to login_url
+      end
+    end
```
Comment (maelfosso) [inline private-events/app/controllers/users_controller.rb]: That method is repeated. I saw it in **EventController** too. 
Please, could you move it in **ApplicationController**

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @Ashah15 @MarvellousUbani 

THank you for that project. I really loved the design :smile_cat: 
However, there are some features not implemented
1. According to that [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png), the user's show page displays events the user has created and events that the user is attending to separated in past and upcoming
**Here is the list of created event**. You will notice some created by **_Alice_**
Screenshot from 2019-12-31 10-36-50
But if I **open the user (alice) show page**, you won't see anything
Screenshot from 2019-12-31 10-37-14

2. You should write integrations tests. There are none tests in your code.

3. Read the comments below.


Kindly make the changes and submit a new code review request.
Thank you.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Nice work so far but the following issues need to be fixed ⬇️** 

- The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations)).

- You need to write integration tests for all the futures.

- Implement the changes from the previous code review.

- Set up Stickler to lint your code, find out more on how to do that in [here](https://github.com/microverseinc/linters-config).

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

File: private-events/app/controllers/events_controller.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
+class EventsController < ApplicationController
+	before_action :logged_in_user, only: [:new, :create]
```
Comment (maelfosso) [inline private-events/app/controllers/events_controller.rb]: I think you should add show here, otherwise there will be an error.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# :wave: :wave:  Hello I can see you did some Improvements based on the review, lets take a look eye_speech_bubble shall we.

- According to that [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) to see the detail of the event, the user should be logged in, please fix it.

- Also you forgot to write integration tests for Event. And about testing, they should write using Rspec. Please, check that in that [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png)
So, please rewrite them all using RSpec.

- There is that error when trying to see the event details from the home page when the user is not connected
Screenshot from 2020-01-07 10-24-01

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: spec/features/create_event_spec.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	RSpec.describe 'Creating an event', type: :feature do
+	  scenario 'valid inputs' do
+	  	login_with 'mike'
+	    visit new_event_path
+	    fill_in "event_name", with: "Miss"
+	    fill_in "event_event_date", with: DateTime.current.strftime("%d/%m/%Y")
+	    click_on "Save changes"
+	    visit events_path
+	    expect(page).to have_content("Miss")
+	  end
```
Comment (samgaco) [inline spec/features/create_event_spec.rb:12]: :+1:

**Review by samgaco — APPROVED** [review]
Everything corrected accordingly, you can move on :+1: 

------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**